File: code/functions/javascript/j_float/wasm.js
import {wasm_benchmark} from "../bench/wasm_bench.js";
import * as env from "wasi:cli/environment@0.2.0";
import {main} from "./float.js";

/**
 * Wasm specific implementation for the benchmark using WASI:
 * Implements the WASI run-Interface to be called by Wasmtime CLI.
 */
export const run = {
    run() {
        wasm_benchmark(() => {
            // calls benchmark using the WASI API to get program arguments from environment
            // as "process.argv" is not available using ComponentizeJS
            const args = env.getArguments().slice(1);
            if (args.length < 1) {
                console.error("Pass a number as argument.");
                return;
            }
            const n = Number(args[0]);
            if (Number.isNaN(n)) {
                console.error(`Error: '${args[0]}' is not a valid number`);
                return;
            }
            main(n)
        })
    }
}

File: code/functions/python/p_mst/wasm.py
import sys
from unittest.mock import Mock

import bench

# gzip and zlib are not available in componentize-py environment
# for more see: https://github.com/bytecodealliance/componentize-py/issues/96
# and: https://opensource.snarky.ca/Python/WASI#Plans
sys.modules['bz2'] = Mock()
sys.modules['zlib'] = Mock()

from p_mst import main


# Wasm specific implementation for the benchmark using WASI:
# Implements the WASI run-Interface to be called by Wasmtime CLI.
class Run:
    def run(self) -> None:
        bench.benchmark(main)
